addons/binding/org.openhab.binding.onewire/src/main/java/org/openhab/binding/onewire/internal/device/DS2438.java

```diff
@@ -270,13 +276,15 @@ public void refresh(OwBaseBridgeHandler bridgeHandler, Boolean forcedRefresh) th
                         }
                         break;
                     case IBUTTONLINK:
-                        light = bridgeHandler.readDecimalType(sensorId, voltageParameter);
-                        if (light instanceof DecimalType) {
-                            light = new QuantityType<Illuminance>(
-                                    Math.pow(10, (65 / 7.5) - (47 / 7.5) * (Vcc / ((DecimalType) light).doubleValue())),
-                                    SmartHomeUnits.LUX);
-                            callback.postUpdate(CHANNEL_LIGHT, light);
+                        double measured = ((DecimalType) bridgeHandler.readDecimalType(sensorId, voltageParameter))
+                                .doubleValue();
+                        if (measured > 10.0) {
+                            // workaround bug in DS2438
+                            measured = 0.0;
                         }
+                        light = new QuantityType<Illuminance>(Math.pow(10, (65 / 7.5) - (47 / 7.5) * (Vcc / measured)),
```

Comment:
The division by `measured` will result in an `ArithmeticException` "division by zero`.

Reply:
The formula is correct. I‘ll add a range-check for the measured value (>0 && <=10).